### PR TITLE
fix(DTFS2-7091): Handle invalid TFM-UI user session

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Ensure TFM-UI invalid user session does not crash the container

## Resolution :heavy_check_mark:
* Deal and user session validation, if void user will be re-directed to problem with service page.

## Miscellaneous :heavy_plus_sign:
* Added few logs for troubleshooting.
* Improved test coverage.
* Lint fixes.

### CODE REVIEW

File: portal-api/src/v1/controllers/deal-status.controller.js [hunk 1]
```diff
@@ -66,6 +66,8 @@ exports.update = async (req, res) => {
 
     const currentStatus = deal.status;
 
+        console.info('Updating portal deal %s status to %s from %s', dealId, newStatus, currentStatus);
```
Comment (AlexBramhill) [inline portal-api/src/v1/controllers/deal-status.controller.js:69]: fwiw - over indented?

**Review by AlexBramhill — DISMISSED** [review]

File: portal-api/src/v1/controllers/deal-status.controller.js (hunk 1 above)
Comment (abhi-markan) [reply portal-api/src/v1/controllers/deal-status.controller.js:69]: For troubleshooting with race condition.

File: trade-finance-manager-ui/server/controllers/case/underwriting/index.js [hunk 2]
```diff
@@ -20,11 +21,21 @@ const { hasAmendmentInProgressDealStage, amendmentsInProgressByDeal } = require(
  * renders underwriting page with deal, tfm, dealId, user, leadUnderwriter, pricingAndRisk, underwriterManagersDecision
  */
 const getUnderwriterPage = async (req, res) => {
-  const dealId = req.params._id;
+  const dealId = req?.params?._id;
   const { user, userToken } = req.session;
 
+  if (!dealId || !user || !userToken) {
```
Comment (AlexBramhill) [inline trade-finance-manager-ui/server/controllers/case/underwriting/index.js:27]: Does our tfm middleware not handle having a session already (am assuming it's like portal) :) (so wouldn't need user token or user check here?)?

File: trade-finance-manager-ui/server/controllers/case/underwriting/lead-underwriter/index.js [hunk 3]
```diff
@@ -8,7 +8,12 @@ const getLeadUnderwriter = async (deal, user, token) => {
   let currentLeadUnderWriter;
   let currentLeadUnderWriterUserId;
 
-  if (deal.tfm.leadUnderwriter) {
+  if (!deal || !user || !token) {
```
Comment (AlexBramhill) [inline trade-finance-manager-ui/server/controllers/case/underwriting/lead-underwriter/index.js:11]: same re user and token :)

**Pull request merged**